modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -690,3 +696,171 @@ PinMeshGenerator::generate()
 
   return std::move(*_build_mesh);
 }
+
+std::unique_ptr<CSG::CSGBase>
+PinMeshGenerator::generateCSG()
+{
+  auto csg_obj = std::move(getCSGBaseByName(_reactor_params));
```

Comment:
so if we have two pins this wont work?

Reply:
Good point, I still need to figure out how to handle this for the case when two pins are defined